### DISCUSSION TITLE
Difference between two types of gates in a quest?

### DISCUSSION THREAD

**Discussion by ksetdekov** [streetcomplete/StreetComplete#3729] [Q&A] [answered]
In a quest: "How does this road cross the barrier here?" We can see two answers labeled "gate" with very similar pictures.

How does one determine which is relevant where and what tags are used ultimately got them?

I would be grateful if someone more experienced with code could share their insight.

![Screenshot_20220207-084837](https://user-images.githubusercontent.com/16122863/152731896-b8228611-4147-4830-8e1a-14885e03daa5.png)

**Comment by matkoniecz** [comment]
See https://github.com/streetcomplete/StreetComplete/issues/3717

There is no difference.

Listing one is tiny gate for pedestrian, one is large gate for cars is intentional. As people without OSM experience would be unlikely to tag pedestrian gates as vehicle gate (or vehicle gates as pedestrian gate).

The same name is also intentional to hint that there is no actual difference between them.

I tried fitting both on one combined image but it was not legible.

I considered having longer label with explanation but had no idea about anything clear and not filing entire image.

-----------

I am not entirely happy about the current solution, but I had no better idea. Pull requests or mockups are welcome.

I even thought about tagging `maxwidth=small` or something similar for small ones and having distinct names but AFAIK there is no accepted standard for that. Establishing some standard for that would be also welcomed, such distinction may be useful for routing - especially for emergency vehicles. See https://wiki.openstreetmap.org/wiki/Tagging_mailing_list (maybe in addition https://wiki.openstreetmap.org/wiki/Proposal_process ) may be useful tools for confirming that such extra tagging is welcome.

**Reply by eginhard** [reply]
The smaller one could be labelled "Pedestrian gate"?

**Reply by matkoniecz** [reply]
(1) anyone looking at applied tags would be even more confused

(2) "The same name is also intentional to hint that there is no actual difference between them."

Though maybe assuming that minority would look at tags so confusing them is preferable may be the correct answer?

**Comment by mnalis** [comment]
@matkoniecz Maybe call small one _"Pedestrian gate"_ and tag it additionally with `foot=yes`. 

That way IMHO it would be less confusing than two different pictures with same description, and would also be tagged differently so people familiar with OSM tags would be understanding too.

**Reply by matkoniecz** [reply]
> Maybe call small one "Pedestrian gate" and tag it additionally with foot=yes.

That would be incorrect for many such gates that are private and not usable by random people.

**Reply by mnalis** [reply]
That is true! Than, mark _"Pedestrian gate"_ with `motor_vehicle=no` instead. That should be always correct, and does not imply that other traffic is allowed, just that motorized ones isn't.

**Reply by matkoniecz** [reply]
What about motorcycles? It is not typical but I seen some places where such small gates were gates to motorcycle parkings?

Though maybe it would be covered by "Pedestrian gate" name?

**Reply by mnalis** [reply]
That would seem to be quite special case, so user should probably leave a note. (Because, it might need even further refinings, as it might be for example wide enough for [mofa=yes](https://wiki.openstreetmap.org/wiki/Key:mofa) and/or  [moped=yes](https://wiki.openstreetmap.org/wiki/Key:moped) but [motorcycle=no](https://wiki.openstreetmap.org/wiki/Key:motorcycle)). Also  [width=*](https://wiki.openstreetmap.org/wiki/Key:width) and/or [motorcycle:conditional=*](https://wiki.openstreetmap.org/wiki/Key:motorcycle:conditional) might be beneficial is such special cases, which further recommends leaving a note. Or a user can choose other (non-pedestrian) _"gate"_ answer if that is good enough.

As well, _"Pedestrian gate"_ name as suggested by @eginhard seems good description to me in combination with `motor_vehicle=no`.

**Reply by matkoniecz** [reply]
That makes sense to me, but I will leave it up for a bit and also ask for additional feedback elsewhere before implementing.

(and anyway it is a new text, so it would wait for v41 anyway)

My main worry is that when there is `barrier=gate` on footway leading to private home it is rather `motor_vehicle=private` as homeowner may use it also to drive motorcycle through it.

**Reply by HolgerJeromin** [reply]
We could tag `motorcar=no` which would not lie about motorcycles...

**Reply by mnalis** [reply]
Problem with `motorcar=no` is it is only about automobiles, so we'd also need `goods=no`, `hgv=no`, `motorhome=no`, `coach=no`, `agricultural=no`,  etc. to indicate other four-wheeled vehicles - see https://wiki.openstreetmap.org/wiki/Key:access#Land-based_transportation

**Comment by smichel17** [comment]
The current behavior is problematic because it breaks SC's pattern of having distinct/meaningful answers. As a mapper, I agonize over picking the right value, so I am not contributing bad data to OSM. Scrutinizing the pictures to figure out which is the right one, when in fact they are going to tag the same thing, is a **waste of time.**

Any solution that differentiates the two while producing the same tagging is a problem for the same reason. When something sits on the line between the two categories, I am going to hem and haw about which answer to pick (and maybe open a note), when in reality it did not matter.

- If the extra tagging of `motorcar=no` or similar is useful data, then I support the solution which adds that tag and clarifies the text for the two "gate" answers.

- If the extra tagging is just so that we can find something different to tag for the two answers, then I support removing one answer and finding a different way of showing multiple pictures for one answer (e.g. https://github.com/streetcomplete/StreetComplete/discussions/3586)

**Reply by 10992-osm** [reply]
I was thinking along the lines of your second suggestion, in fact I had started writing before I saw your post, which I put below:

If the two pictures can't be combined into one standard square, could the two pictures be combined into one double-width button, so that it's clear that they are both the same? (No white gap, the label only appearing once, the whole double-width item being highlighted when selected.)

If double-width is unsuitable, you could always try full (triple) width.

Question:
Is the picture grid always 3 across, is it always at least 2 across?

**Reply by mnalis** [reply]
True; if the difference provided by extra tag is not useful enough (or is too problematic), and merging pictures into one answer is not doable (nor is finding picture that is halfway between those two gates), then I'd also think removing one of the answers is probably the best. 

I would then likely remove pedestrian gate - one of the reasons being that as noted before it is less likely to be useful, not only as it available for less forms of transport, but also as very often it goes to private property (where leaving a note saying "private" would actually be more useful than answering the quest with specific type of private barrier)

**Comment by matkoniecz** [comment, marked as answer]
This problem was reported in one more place. Thanks that it was reported also here, without that I would likely consider it as an isolated problem.

The plan is to show only larger gate, but with description "Gate, any size"

--------

EDIT: Changed now

**Reply by matkoniecz** [reply]
Now I am also considering "gate, pedestrian gate"

Any opinion about that?

**Reply by mnalis** [reply]
> The plan is to show only larger gate, but with description "Gate, any size"

I'm OK with that as solution too

> Now I am also considering "gate, pedestrian gate"

you mean, in addition to that "Gate, any size", or instead of it? In any case I like single "Gate, any size" better.

(Interestingly, I never got those posts on email, even if I'm subscribed to it. Maybe github does not send e-mail notifications once the discussion is marked as answer? Of some other foobar...)

**Reply by 10992-osm** [reply]
> > The plan is to show only larger gate, but with description "Gate, any size"
> 
> I'm OK with that as solution too
> 
> > Now I am also considering "gate, pedestrian gate"
> 
> you mean, in addition to that "Gate, any size", or instead of it? In any case I like single "Gate, any size" better.
> 
> (Interestingly, I never got those posts on email, even if I'm subscribed to it. Maybe github does not send e-mail notifications once the discussion is marked as answer? Of some other foobar...)

@mnalis I received a notification of your comment by e-mail…

**Reply by mnalis** [reply]
thanks for the confirmation @10992-osm. I've also received e-mail notification for your message here. I'm out of ideas why I didn't receive it then. Perhaps a temporary glitch in the matrix... Or somehow I missed it.
